lib/openhab/core/events/timer_event.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module OpenHAB
+  module Core
+    module Events
+      # @deprecated OH3.4 this guard is not needed on OH4
+      if Gem::Version.new(OpenHAB::Core::VERSION) >= Gem::Version.new("4.0.0")
+        java_import org.openhab.core.automation.events.TimerEvent
+
+        #
+        # Adds methods to core openHAB TimerEvent to make it more natural in Ruby
+        #
+        # This event can be triggered by a `DateTimeTrigger`, `cron`, or `TimeOfDay` trigger.
+        #
+        # @since openHAB 4.0
+        #
+        class TimerEvent < AbstractEvent
+          #
+          # @!attribute [r] cron_expression
+          # @return [String, nil] The cron expression that triggered this event.
+          #
+          def cron_expression
+            payload&.dig(:cronExpression)
```

Comment:
Why is `&.[](:cronExpression)` preferred to using dig?

Reply:
Because you're only going a single level. It skips a method call (dig will call [] internally). Dig is nice when digging into a deeply nested data structure, especially if it's possible part of it will be missing, but is just extra indirection for a single level.